presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPageSourceProvider.java

```diff
@@ -116,6 +115,16 @@ public ConnectorPageSource createPageSource(
                     false);
         }
         else if (tableType == HudiTableType.MOR) {
+            // use first log file as base file for MOR table if it hasn't base file
+            if (!hudiSplit.getBaseFile().isPresent()) {
+                hudiSplit = new HudiSplit(hudiSplit.getTable(),
+                        hudiSplit.getInstantTime(),
+                        hudiSplit.getPartition(),
+                        Optional.of(hudiSplit.getLogFiles().get(0)),
+                        hudiSplit.getLogFiles(),
+                        hudiSplit.getAddresses(),
+                        hudiSplit.getNodeSelectionStrategy());
+            }
```

Comment:
`
HudiFile baseFile = split.getBaseFile().orElseThrow(
        () -> new PrestoException(HUDI_CANNOT_OPEN_SPLIT, "Split without base file not supported yet"));
Path path = new Path(baseFile.getPath());
`
it needs to be replaced twice. 

https://github.com/prestodb/presto/blob/d83b10737eab1b448c09f11061ce195947ad3666/presto-hudi/src/main/java/com/facebook/presto/hudi/HudiRecordCursors.java#L74

https://github.com/prestodb/presto/blob/d83b10737eab1b448c09f11061ce195947ad3666/presto-hudi/src/main/java/com/facebook/presto/hudi/HudiRecordCursors.java#L112

Maybe it can be avoided when building, or keep current

Reply:
Got it. We could extract a method to reduce the repeated code.